### PR TITLE
Fix writtenIntermediateBytes metric to capture correct intermediate w…

### PR DESCRIPTION
**Pull request by jaystarshot**
The writtenIntermediateBytes metric was noop because plan.isOutputTableWriterFragment() will always be true and wrong metrics were updated when there are temporary table writes currently since there was a bug in plan Fragmenter where it would pass temporary tables' writer operator as output table writes hence populating the wrong metric.
code [link](https://github.com/prestodb/presto/blob/4214d3826d2e49c580a00974a704685045f79bf7/presto-main/src/main/java/com/facebook/presto/execution/QueryStats.java#L389).

Fixed by adding the information (whether temporary) to the tablewriterNode and filtering out those

## Impact
Fixed writtenIntermediateBytes metric. Users of CTE and Exchange materialization should now be able to see the correct populated metric

## Test Plan
Production tested

## Contributor checklist

- [ ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

General Changes
* Fixed writtenIntermediateBytes metric by ensuring its correct population through temporary table writers.

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java [hunk 1]
```diff
@@ -81,7 +81,7 @@ public SubPlan createSubPlans(Session session, Plan plan, boolean forceSingleNod
                 sqlParser,
                 idAllocator,
                 variableAllocator,
-                getTableWriterNodeIds(plan.getRoot()));
```
Comment (jaystarshot) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java:84]: Main fix

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/CteUtils.java [hunk 2]
```diff
@@ -25,6 +25,7 @@
 
 public class CteUtils
 {
+    public static final String TEMPORARY_TABLE_SCHEMA = "temporary_table_schema";
```
Comment (feilong-liu) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/CteUtils.java:28]: where is this used?

Comment (jaystarshot) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/CteUtils.java]: Removed thanks

**Review by feilong-liu — APPROVED** [review]
LGTM. I also briefly checked the code, looks like that the "outputTableWriterFragment" variable is only used for accounting purpose.

Can you maybe add a comment for the "outputTableWriterFragment" member in "PlanFragment" class stating that this is true only for final output fragment, but not for temporary table writer?

**Comment by jaystarshot** [comment]
cc: @tdcmeehan

**Review by tdcmeehan — APPROVED** [review]

**Comment by steveburnett** [comment]
Nit (commenting because it's not enough for a review): in Description, suggest deleting the lines in the release note entry after the line `Fixed writtenIntermediateBytes metric...`

**Comment by steveburnett** [comment]
> Nit (commenting because it's not enough for a review): in Description, suggest deleting the lines in the release note entry after the line `Fixed writtenIntermediateBytes metric...`

Thanks!

**Pull request merged**